src/orion/core/io/convert.py

```diff
@@ -122,6 +125,126 @@ def generate(self, filepath, data):
             self.json.dump(data, f)
 
 
+class GenericConverter(BaseConverter):
+    """Generic converter for any configuration file type.
+
+    For each parameter dimension declared here, one must necessarily
+    provide a ``name`` keyword inside the `Dimension` building expression.
+
+    Implementation details: As this class is supposed to provide with a
+    generic text parser, semantics are going to be tied to their consequent
+    usage. A template document is going to be created on `parse` and filled
+    with values on `read`. This template document consists the state of this
+    `Converter` object.
+
+    Dimension should be defined for instance as:
+    ``meaningful_name~uniform(0, 4)``
+
+    """
+
+    def __init__(self, regex=r'([\/]?[\w|\/|-]+)~([\+]?.*\)|\-|\>[A-Za-z_]\w*)',
+                 expression_prefix=''):
+        """Initialize with the regex expression which will be searched for
+        to define a `Dimension`.
+        """
+        self.re_module = importlib.import_module('re')
+        self.regex = self.re_module.compile(regex)
+        self.expression_prefix = expression_prefix
+        self.template = None
+        self.has_leading = defaultdict(str)
+        self.conflict_msg = "Namespace conflict in configuration file '{}', under '{}'"
+
+    def _raise_conflict(self, path, namespace):
+        raise ValueError(self.conflict_msg.format(path, namespace))
+
+    def parse(self, filepath):
+        r"""Read dictionary out of the configuration file.
+
+        Create a template for Python 3 string format and save it as this
+        object's state, by substituing '{\1}' wherever the pattern
+        was matched. By default, the first matched group (\1) corresponds
+        with a dimension's namespace.
+
+        .. note:: Namespace in substitution templates does not contain the first '/'.
+
+        Parameters
+        ----------
+        filepath : str
+           Full path to the original user script's configuration.
+
+        """
+        with open(filepath) as f:
+            self.template = f.read()
+
+        # Search for Oríon semantic pattern
+        pairs = self.regex.findall(self.template)
+        ret = dict(pairs)
+
+        # Every namespace given should be unique,
+        # raise conflict if there are duplicates
+        if len(pairs) != len(ret):
+            namespaces = list(zip(*pairs))[0]
+            for name in namespaces:
+                if namespaces.count(name) != 1:
+                    self._raise_conflict(filepath, name)
+
+        # Create template using each namespace as format key,
+        # exactly as provided by the user
+        subst = self.re_module.sub(r'{', r'{{', self.template)
+        subst = self.re_module.sub(r'}', r'}}', subst)
+        substituted, num_subs = self.regex.subn(r'{\1!s}', subst)
+        assert len(ret) == num_subs, "original: {}\n, regex:{}".format(self.template, self.regex)
```

Comment:
Can we replace with a if; raise? I would limit assertions to unit-tests

Reply:
I am not sure, that is what Python's `assert` originally was intended to be used for right?

Reply:
As far as I understand assert is meant for debugging on the developer's side.

https://docs.python.org/3.5/reference/simple_stmts.html#assert

In this case it would be used by the user to debug it's configuration file definition according to parsing errors.

Reply:
Oh yes this assertion was meant to be for the developer. If this doesn't work, something was wrong with parsing code. It is not meant to tell the user that they did anything wrong, but to potentially catch a faulty runtime behavior cause by a case that I, the developer, did not expect and hence did not write a test for.

I prefer to remove it completely, than to substitute it with another way to have a debugging line.

Reply:
I don't understand your last sentence: 
> I prefer to remove it completely, than to substitute it with another way to have a debugging line.

Then I would be fine with an assertion, but a message should tell the developer what kind of problem is there.